cinema/views.py

```diff
@@ -1,45 +1,109 @@
-from rest_framework.decorators import api_view
+from rest_framework import status, generics, mixins, viewsets
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
 from rest_framework.response import Response
-from rest_framework import status
+from django.http import HttpRequest
 
-from django.shortcuts import get_object_or_404
+from cinema.models import Movie, Actor, Genre, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    CinemaHallSerializer
+)
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
 
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
```

Comment:
Using PrimaryKeyRelatedField in your serializer you will have N+1 query problem for genres and actors.

Reply:
Use `prefetch_related` to solve N+1 problem.